src/main/java/bsise/server/reply/ReplyService.java

```diff
@@ -90,10 +93,13 @@ public List<ReplyResponseDto> findMyLetterAndReply(UUID userId, Integer size) {
                 .toList();
     }
 
-    public Page<ReplyResponseDto> findMyLetterAndReply(UUID userId, Pageable pageable) {
+    public Page<ReplyResponseDto> findMyLetterAndReply(UUID userId, int year, Pageable pageable) {
         validateUserId(userId);
 
-        Page<Reply> replies = replyRepository.findRepliesByOrderByCreatedAt(userId, pageable);
+        LocalDateTime startOfYear = Year.of(year).atDay(1).atStartOfDay();
+        LocalDateTime nextStartOfYear = startOfYear.plusYears(1);
+
+        Page<Reply> replies = replyRepository.findRepliesByOrderByCreatedAt(userId, startOfYear, nextStartOfYear, pageable);
```

Comment:
`year` 값으로 2024를 받으면 `2024-01-01 - 2025-01-01` 사이의 편지들을 조회하는 것으로 오해될 소지가 있는 것 같습니다.
아래 코드로 제안 드려요

```suggestion
        LocalDate startOfYear = Year.of(year).atDay(1);
        LocalDate endOfYear = startOfYear.with(TemporalAdjusters.lastDayOfYear());

        Page<Reply> replies = replyRepository.findRepliesByOrderByCreatedAt(userId, LocalDateTime.of(startOfYear, LocalTime.MIN), LocalDateTime.of(endOfYear, LocalTime.MAX), pageable);
```

Reply:
음... 원래 제안주신 코드대로 구현했다가, LocalTime.MAX와 `<=` 비교를 해도 혹시나 23:59:59.999999... 이시점에 작성된 게 포함 안되면 어쩌지 하는 생각이 들어서 jpql의 between을 대신해 `createdAt >= startOfYear AND createdAt < nextStartOfYear` 이렇게 변경했는데요.

사실 저는 오해의 소지가 그렇게 있나? 싶기는 해요.
하지만 다른 부분 코드와 통일성이 있어야 할 것 같기는 해서... 변경해서 커밋해두겠습니다

Reply:
`nextStartOfYear` 변수명이 "어떤 연도의 다음 시작일(?)"이라고 받아들여졌습니다
차라리 `startOfNextYear` 이면 "다음 연도의 시작일(1월 1일)"가 나을 수 있겠네요..!
+  `startOfNextYear`라고 해도 다음 해 1월 1일까지 데이터를 가져오는 것 같아서 `endOfYear`로 맞췄습니다!

Reply:
근데 제가 예전에 만든 LetterRepository에도 조건이 샤피랑 똑같네요 ;;
과거의 나는 왜그랬을까..

Reply:
ㅋㅋㅋ 수정 중입니다~ 완료하고 바로 PR 할게요!